src/Model/Timeframe.php

```diff
@@ -688,6 +689,12 @@ public function getGrid(): int {
 	 */
 
     public function hasTimeframeTimeOverlap( Timeframe $otherTimeframe ) {
+
+		// If one of the timeframes is a full day timeframe, there will always be a time overlap
```

Comment:
Das stimmt vielleicht mit den Annahmen im Hinterkopf, für die Verwendung innerhalb der `overlaps` Methode, aber da es sich um eine public Methode handelt, kann es auch Timeframes geben, für die beide `fullDay == true` gilt, aber die nicht überlappen.

Reply:
Das würde dann aber auch für nicht fullDay Zeitrahmen gelten. Die hasTimeframeTimeOverlap schaut nur, ob sich die Stunden an einem Tag überlappen und orientiert sich dabei am end-time und start-time Postmeta. Hier ist zum Beispiel mal das end-time Postmeta:

![image](https://github.com/wielebenwir/commonsbooking/assets/6433480/f3ac63a9-5d96-4fba-9130-f479ccdd4619)

Dieser "Shortcut" ist nur dafür da um diesen Case abzufangen, ich müsste es mal testen aber wenn man den weglässt sollte trotzdem alles funktionieren.

Sprich: Die hasTimeframeTimeOverlap würde bei zwei Timeframes
13.11.2023 08:00 - 09:00
15.11.2023 08:30 - 09:30

true zurückgeben, weil sie sich in den Abhol-und Rückgabezeitpunkten überlappen. Für das Datum ist die hasTimeframeDateOverlap zuständig.

Reply:
Und vielleicht noch einen Verweis zu overlaps ... also das man `overlaps` für eine erschöpfendes Matching benutzen soll.